server/src/main/java/org/hyperskill/community/flashcards/card/model/Card.java

```diff
@@ -0,0 +1,39 @@
+package org.hyperskill.community.flashcards.card.model;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+import org.springframework.data.annotation.CreatedDate;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+import java.time.Instant;
+
+/**
+ * Basic class for all flashcards, contains fields shared among all subclasses
+ */
+@Getter
+@Setter
+@Document
+@ToString
+public abstract class Card {
+    @Id
```

Comment:
Wouldn't card be a perfect use case for a sealed class ... permits MultipleChoiceQuiz, SingleChoiceQuiz, QuestionAndAnswerCard

Then we could make perfect use of Java 21 pattern matching also.

Reply:
Agreed, changed to a sealed class with three subclasses.